dynawo/sources/Models/Modelica/Dynawo/NonElectrical/Blocks/NonLinear/FixedBooleanDelay.mo

```diff
@@ -13,27 +13,35 @@ within Dynawo.NonElectrical.Blocks.NonLinear;
 */
 
 block FixedBooleanDelay "Delay block with fixed delay time for boolean input"
-
   import Modelica;
   import Dynawo;
 
   extends Modelica.Blocks.Interfaces.BooleanSISO(y(start = Y0));
 
-  parameter Dynawo.Types.Time tDelay "Delay time of output with respect to input signal, in s";
+  parameter Dynawo.Types.Time tDelay "Delay time in s";
+
   parameter Boolean Y0 "Initial value of output";
 
 protected
-  Integer yInteger;
+  Real uReal "Boolean input signal converted to real";
+  Real yReal "Delayed real signal";
 
 equation
-  yInteger = floor(delay(if u then 1 else 0, tDelay));
+  uReal = if u then 1 else 0;
+  yReal = delay(uReal, tDelay);
```

Comment:
I don't really understand why this works while the same thing with a floor after the delay doesn't, knowing that floor is supposed to be working with dynawo. @rosiereflo and @adriguir any clue ?